bionic/decorators.py

```diff
@@ -103,7 +103,33 @@ def memoize(enabled):
     if not isinstance(enabled, bool):
         raise ValueError(f"Argument must be a boolean; got {enabled!r}")
 
-    return provider_wrapper(AttrUpdateProvider, 'should_memoize', enabled)
+    return provider_wrapper(AttrUpdateProvider, '_should_memoize', enabled)
+
+
+def volatile(enabled):
```

Comment:
This should probably default to `True`. In fact, it's not obvious to me that we need a parameter at all, although I can see why you wanted to maintain consistency.

Reply:
I initially didn't add a parameter to this decorator, so I had to default the value of `is_volatile` to `False` in `AttributeProvider`.  But when the decorator update the attribute, it failed the check where an attribute can be set only once in `AttrUpdateProvider`. It was easier to keep the enable variable like other attribute decorators.

Reply:
I'm probably missing some more context here. I put a default value on the attribute since the attribute should be false by default. Without having a default attribute value, it won't be possible to set the variable to false unless we have another method to set it to false aside from this decorator.

Reply:
Ah, yes, that's also a decent reason to keep the parameter. But we should probably allow `@volatile` to be used like this:

```
@builder
@bn.volatile # no arguments!
def entity(...):
    ....
```

`@pyplot` is an existing example of a decorator that can be used with or without arguments.

Reply:
Oh I didn't realize there was an easy way to do this!